src/ray/object_manager/plasma/store.cc

```diff
@@ -89,11 +89,15 @@ PlasmaStore::PlasmaStore(instrumented_io_context &main_service, IAllocator &allo
           spill_objects_callback, object_store_full_callback,
           /*get_time=*/
           []() { return absl::GetCurrentTimeNanos(); },
-          [this]() { return GetDebugDump(); }),
+          // absl can't check thread safety for lambda
+          [this]() ABSL_NO_THREAD_SAFETY_ANALYSIS { return GetDebugDump(); }),
+      total_consumed_bytes_(0),
       get_request_queue_(io_context_, object_lifecycle_mgr_,
-                         [this](const ObjectID &object_id, const auto &request) {
-                           this->AddToClientObjectIds(object_id, request->client);
-                         },
+                         // absl failed to check thread safety for lambda
+                         [this](const ObjectID &object_id, const auto &request)
+                             ABSL_NO_THREAD_SAFETY_ANALYSIS {
```

Comment:
cc @mwtian I can't work around the annotation with lambda and my conclusion is absl can't analyze thread safety for lambdas (which makes sense)

Reply:
Yes I remember there were issues with lambda and thread safety analysis. Maybe higher versions of clang can help but not 100% sure.